wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/BaseFactoryParameterResolver.java

```diff
@@ -60,7 +60,12 @@ protected FactoryMetaDto createFactory(
             contentProvider,
             extractOverrideParams(factoryParameters),
             getSkipAuthorisation(factoryParameters))
-        .orElseGet(() -> newDto(FactoryDto.class).withV(CURRENT_VERSION).withSource("repo"))
+        .orElseGet(
+            () ->
+                newDto(FactoryDevfileV2Dto.class)
+                    .withDevfile(Map.of("schemaVersion", "2.3.0"))
+                    .withV(CURRENT_VERSION)
```

Comment:
what is the CURRENT_VERSION stand for ?

Reply:
This is Che specific factory API version constant. `CURRENT_VERSION = "4.0"`